freelist.go

```diff
@@ -389,5 +389,29 @@ func (f *freelist) reindex() {
 // arrayMergeSpans try to merge list of pages(represented by pgids) with existing spans but using array
 func (f *freelist) arrayMergeSpans(ids common.Pgids) {
 	sort.Sort(ids)
+	common.Verify(func() {
+		idsIdx := make(map[common.Pgid]struct{})
+		for _, id := range f.ids {
+			// The existing f.ids shouldn't have duplicated free ID.
+			if _, ok := idsIdx[id]; ok {
+				panic(fmt.Sprintf("detected duplicated free page ID: %d in existing f.ids: %v", id, f.ids))
+			}
+			idsIdx[id] = struct{}{}
+		}
+
+		prev := common.Pgid(0)
+		for _, id := range ids {
+			// The ids shouldn't have duplicated free ID.
+			if prev == id {
```

Comment:
maybe leave a little comment here that the meta page 0 can never be in the free list, otherwise you'll potentially panic here

Reply:
Updated the comment.

Reply:
thx!